main.tf

```diff
@@ -11,12 +11,13 @@ resource "aws_acm_certificate" "default" {
 
 data "aws_route53_zone" "default" {
   count        = "${var.process_domain_validation_options == "true" && var.validation_method == "DNS" ? 1 : 0}"
-  name         = "${var.domain_name}."
+  name         = "${local.zone_name}"
```

Comment:
Hrm... maybe the ending `.` is optional these days?

Reply:
https://serverfault.com/a/803037

Reply:
Oh I did not see your serverfault link. In this case let's remove the dot.

Reply:
for backwards compatibility, `var.domain_name` should not include a dot, it's added in the code.
for consistency, I think it should not be included in `var.zone_name` either.

```
zone_name                 = "${var.zone_name == "" ? var.domain_name. : var.zone_name.}" 
```

Reply:
Okay then I'll just add the dot like this, as @osterman suggested:
```
data "aws_route53_zone" "default" {
  count        = "${var.process_domain_validation_options == "true" && var.validation_method == "DNS" ? 1 : 0}"
  name         = "${local.zone_name}."
  private_zone = false
}

```